### ISSUE TITLE
NTI vs OTI inconsistency in @record missed property handling 

### ISSUE THREAD

**Issue by dmitryxcom** [google/closure-compiler#2663]
To repro, compile this with OTI and NTI:

```javascript
/** @record */
class ChromeWebPushMessage {
  constructor() {
    /** @export {string} */
    this.deviceId;

    /** @export {number} */
    this.timestampSec;

    /** @export {number} */
    this.displayCap;
  }
}

/**
 * @param {!ChromeWebPushMessage} data The payload extracted from the push event.
 * @param {string} url The URL to navigate to if the user clicks the
 *     notification.
 */
function displayNotificationPayload(data, url) {
  if (data.xxy) { // <---------- THIS THING, IS IT WAI NOT TO REPORT MISSING PROPERTY?
    const /** string */ x = data.xxy;
  }
}
```

OTI reports missing property (expected), while NTI does not.

originally filed:
https://github.com/google/closure-compiler-js/issues/75

**Comment by dimvar** [comment]
Yes this is WAI. You are asking if the property exists and then acting on that. NTI doesn't warn for property-existence checks.
